app/src/main/java/com/keylesspalace/tusky/util/LinkHelper.java

```diff
@@ -117,20 +117,63 @@ public void onClick(View widget) {
             /* Add zero-width space after links in end of line to fix its too large hitbox.
              * See also : https://github.com/tuskyapp/Tusky/issues/846
              *            https://github.com/tuskyapp/Tusky/pull/916 */
-            if(end >= builder.length()){
+            if (end >= builder.length() ||
+                    builder.subSequence(end, end + 1).toString().equals("\n")){
                 builder.insert(end, "\u200B");
-            } else {
-                if(builder.subSequence(end, end + 1).toString().equals("\n")){
-                    builder.insert(end, "\u200B");
-                }
             }
-
         }
+
         view.setText(builder);
         view.setLinksClickable(true);
         view.setMovementMethod(LinkMovementMethod.getInstance());
     }
 
+    /**
+     * Put mentions in a piece of text and makes them clickable, associating them with callbacks to
+     * notify when they're clicked.
+     *
+     * @param view the returned text will be put in
+     * @param mentions any '@' mentions which are known to be in the content
+     * @param listener to notify about particular spans that are clicked
+     */
+    public static void setClickableMentions(
```

Comment:
This method could return Spaanable or something like that but some other methods also take TextView :thinking: